src/Raven.Server/Documents/Handlers/Processors/Studio/StudioCollectionHandlerProcessorForDeleteCollection.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Operations;
+using Raven.Server.Documents.Operations;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Web.Studio;
+
+namespace Raven.Server.Documents.Handlers.Processors.Studio
+{
+    internal class StudioCollectionHandlerProcessorForDeleteCollection : AbstractStudioCollectionHandlerProcessorForDeleteCollection<DatabaseRequestHandler, DocumentsOperationContext>
+    {
+        public StudioCollectionHandlerProcessorForDeleteCollection([NotNull] DatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override void DeleteCollection(DocumentsOperationContext context, IDisposable returnContextToPool, string collectionName, HashSet<string> excludeIds,
+            long operationId, OperationCancelToken token)
+        {
+            ExecuteCollectionOperation(
+                (runner, collectionNameParam, options, onProgress, tokenParam) =>
+                    Task.Run(async () => await runner.ExecuteDelete(collectionNameParam, 0, long.MaxValue, options, onProgress, tokenParam)),
+                context, returnContextToPool, OperationType.DeleteByCollection, collectionName, operationId, excludeIds, token);
+        }
+
+        protected override long GetNextOperationId()
+        {
+            return RequestHandler.Database.Operations.GetNextOperationId();
+        }
+
+        private void ExecuteCollectionOperation(
+            Func<CollectionRunner, string, CollectionOperationOptions, Action<IOperationProgress>, OperationCancelToken, Task<IOperationResult>> operation,
+            DocumentsOperationContext docsContext, IDisposable returnContextToPool, OperationType operationType, string collectionName, long operationId,
+            HashSet<string> excludeIds, OperationCancelToken token)
+        {
+            var opToken = RequestHandler.CreateTimeLimitedCollectionOperationToken();
```

Comment:
why do we have 2 tokens here? can you explain?

Reply:
Thought lines 50 + 51 each got a token for different uses but I think it might not be the case after all. Changed